app/helpers/shift_commerce/template_definition_helper.rb

```diff
@@ -1,12 +1,19 @@
 module ShiftCommerce
   module TemplateDefinitionHelper
     def render_template_for(item, *args)
-      tpl = item.template_definition
-      if(tpl.present?)
-        render "template_definitions/#{tpl.reference}", *args
+      if item.template_definition.present?
+        if minimal_layout?
+          render "template_definitions/#{tpl.reference}", *args, layout: false
+        else
+          render "template_definitions/#{tpl.reference}", *args
+        end
```

Comment:
This moves the functionality that allows any request with `?minimal=true` or an `X-PJAX` header to return a template without its layout out of the static pages controller, and helps de-duplicate the functionality.